beets/plugins.py

```diff
@@ -399,11 +399,18 @@ def send(event, **arguments):
     Returns a list of return values from the handlers.
     """
     log.debug(u'Sending event: {0}'.format(event))
+    return_values = []
     for handler in event_handlers()[event]:
         # Don't break legacy plugins if we want to pass more arguments
         argspec = inspect.getargspec(handler).args
         args = dict((k, v) for k, v in arguments.items() if k in argspec)
-        handler(**args)
+        result = handler(**args)
+        if isinstance(result, list):
```

Comment:
This special list handling could bite us when other sequence types are involved, or when separate lists are actually significant. Let's just append all the responses to the return list (although we should probably filter out `None`s for convenience) and let clients join lists if they want that.